### PR TITLE
Update eslint devDependencies, fix errors

### PR DESCRIPTION
**Pull request by lencioni**
Nothing too surprising here, just bumping versions to stay up to date.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,9 +53,9 @@
     "chai": "^3.5.0",
     "enzyme": "^2.4.1",
     "eslint": "^3.3.1",
-    "eslint-config-airbnb": "^10.0.1",
-    "eslint-plugin-import": "^1.14.0",
-    "eslint-plugin-jsx-a11y": "^2.1.0",
+    "eslint-config-airbnb": "^14.0.0",
+    "eslint-plugin-import": "^2.2.0",
+    "eslint-plugin-jsx-a11y": "^3.0.2",
     "eslint-plugin-react": "^6.1.2",
```
Comment (ljharb) [inline package.json]: This line needs updating as well to 6.9.0; `npm ls` should be complaining

File: test/ThemeProvider_test.jsx [hunk 2]
```diff
@@ -9,7 +9,7 @@ describe('<ThemeProvider>', () => {
     const wrapper = shallow(
       <ThemeProvider name="tropical">
         <div>Under the sea</div>
-      </ThemeProvider>
+      </ThemeProvider>,
```
Comment (ljharb) [inline test/ThemeProvider_test.jsx]: Are we using the latest babel-preset-airbnb?

File: package.json [hunk 3]
```diff
@@ -53,9 +53,9 @@
     "chai": "^3.5.0",
     "enzyme": "^2.4.1",
     "eslint": "^3.3.1",
```
Comment (ljharb) [inline package.json]: Also this needs to be bumped

File: package.json (hunk 1 above)
Comment (lencioni) [inline package.json]: The only thing `npm ls` complains about is

> npm ERR! peer dep missing: react-native@*, required by react-native-mock@0.2.9

Which I think I might just ignore.

File: test/ThemeProvider_test.jsx (hunk 2 above)
Comment (lencioni) [inline test/ThemeProvider_test.jsx]: `npm outdated` doesn't report anything, but I can bump some version requirements in package.json.

**Comment by lencioni** [comment]
@ljharb PTAL

File: test/ThemeProvider_test.jsx (hunk 2 above)
Comment (ljharb) [inline test/ThemeProvider_test.jsx]: You may have the right version locally installed, but package.json says 2.0.0 and we're on 2.1.1

**Pull request merged**